src/rest_framework_dso/embedding.py

```diff
@@ -0,0 +1,192 @@
+"""Logic to implement object embedding, and retrieval in a streaming-like fashion.
+
+It's written in an observer/listener style, allowing the embedded data to be determined
+during the rendering. Instead of having to load the main results in memory for analysis,
+the main objects are inspected while they are consumed by the output stream.
+"""
+from typing import Dict, Optional, Union
+
+from rest_framework import serializers
+from rest_framework.exceptions import ParseError, PermissionDenied
+
+from dso_api.dynamic_api.permissions import fetch_scopes_for_model
+from rest_framework_dso.fields import AbstractEmbeddedField, EmbeddedField
+from rest_framework_dso.serializer_helpers import ReturnGenerator, peek_iterable
+
+EmbeddedFieldDict = Dict[str, AbstractEmbeddedField]
+
+
+def get_expanded_fields(  # noqa: C901
+    parent_serializer: serializers.Serializer, fields_to_expand: Union[list, bool]
```

Comment:
The list can be make more specific:
```python
fields_to_expand: Union[List[str], bool]
```
Assuming the fiield names are of type string.

Reply:
Done!